File: src/shared/validators/status_validator.py
from src.shared.domain.enums.status_enum import Status
from src.shared.exceptions.unauthorized_exception import UnauthorizedException


def status_validator(status: Status):
    if status == Status.BLOCKED:
        raise UnauthorizedException("Status bloqueado!")

    if status == Status.WAITING:
        raise UnauthorizedException("Status em espera!")

    if status == Status.DISABLED:
        raise UnauthorizedException("Status em desabilitado!")

File: src/shared/validators/status.validator.ts
import { $Enums } from '@prisma/client';
import { UnauthorizedException } from '../exceptions';

export async function statusValidator(status: $Enums.Status) {
  if (status === $Enums.Status.BLOCKED)
    throw new UnauthorizedException('Status bloqueado!');

  if (status === $Enums.Status.WAITING)
    throw new UnauthorizedException('Status em espera!');

  if (status === $Enums.Status.DISABLED)
    throw new UnauthorizedException('Status em desabilitado!');
}
